src/programs/azimuthal_average/azimuthal_average.cpp

```diff
@@ -0,0 +1,1534 @@
+
+		if (iteration_counter >= max_iterations || (max_shift <= max_shift_convergence_threshold && max_rotation <= max_rotation_convergence_threshold))
+		{
+			wxPrintf("returning, iteration = %li, max_shift = %f, max_rotation = %f, average_shift = %f, average_rotation = %f\n", iteration_counter, max_shift, max_rotation, average_image_x_shift, average_image_psi_rotation);
+			
+			// shift and rotate the final image stack,
+			#pragma omp parallel for default(shared) num_threads(max_threads) private(image_counter)
+			for (image_counter = 0; image_counter < number_of_images; image_counter++)
+			{
+				// rotate and shift input_stack * CTF
+
+				// rotate first
+				input_stack_times_ctf[image_counter].BackwardFFT();
+				input_stack_times_ctf[image_counter].Rotate2DInPlace(current_psi_angles[image_counter], 0.0); // 0.0 pixel size is mask with radius of half-box size
+				input_stack_times_ctf[image_counter].ForwardFFT();
+				// then shift
+				input_stack_times_ctf[image_counter].PhaseShift(current_x_shifts[image_counter], current_y_shifts[image_counter], 0.0);
+
+				// rotate and shift input_stack
+
+				// rotate first
+				input_stack[image_counter].BackwardFFT();
+				input_stack[image_counter].Rotate2DInPlace(current_psi_angles[image_counter], 0.0); // 0.0 pixel size is mask with radius of half-box size
+				input_stack[image_counter].ForwardFFT();
+				// then shift
+				input_stack[image_counter].PhaseShift(current_x_shifts[image_counter], current_y_shifts[image_counter], 0.0);
+			}
+			
+			delete[] current_x_shifts;
+			delete[] current_y_shifts;
+			delete[] current_psi_angles;
+			delete[] buffer_stack;
+
+			if (number_of_frames_for_running_average > 1)
+			{
+				delete[] running_average_stack;
+			}
+			return;
+		}
+		else
+		{
+			wxPrintf("Not. returning, iteration = %li, max_shift = %f, max_rotation = %f, average_shift = %f, average_rotation = %f\n", iteration_counter, max_shift, max_rotation, average_image_x_shift, average_image_psi_rotation);
+		}
+
+		// going to be doing another round so we need to make the new sum..
+
+		sum_of_images.SetToConstant(0.0);
+
+		for (image_counter = 0; image_counter < number_of_images; image_counter++)
+		{
+			sum_of_images.AddImage(&buffer_stack[image_counter]);
+		}
+		// vertically-averaged CTF-corrected sum
+		divide_by_ctf_sum_of_squares(&sum_of_images, ctf_sum_of_squares);
+		sum_of_images.BackwardFFT();
+		sum_image_direction(&sum_of_images, 2);
+		sum_of_images.ForwardFFT();
+
+		// print out current alignment reference
+		if (iteration_counter == 1)
+		{
+			sum_of_images.QuickAndDirtyWriteSlice("my_aligned_sum_1.mrc", 1);
+		}
+		if (iteration_counter == 2)
+		{
+			sum_of_images.QuickAndDirtyWriteSlice("my_aligned_sum_2.mrc", 1);
+		}
+		if(iteration_counter == 3)
+		{
+			sum_of_images.QuickAndDirtyWriteSlice("my_aligned_sum_3.mrc", 1);
+		}
+		//
+	} // end alignment cycle
+}
+
+// computes the 1D rotational average and sets the corner values to the average value at edge-1
```

Comment:
Maybe double check that you aren't duplicating functionality already provided via 

Image::Compute1DRotationalAverage